Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -2648,17 +2658,27 @@ internal void AddAvailableDevicesThatAreNowRecognized()
                     continue;
 
                 var layout = TryFindMatchingControlLayout(ref m_AvailableDevices[i].description, id);
-                if (IsDeviceLayoutMarkedAsSupportedInSettings(layout))
+                if (!IsDeviceLayoutMarkedAsSupportedInSettings(layout)) continue;
+
+                if (layout.IsEmpty())
                 {
-                    try
-                    {
-                        AddDevice(m_AvailableDevices[i].description, false,
-                            deviceId: id,
-                            deviceFlags: m_AvailableDevices[i].isNative ? InputDevice.DeviceFlags.Native : 0);
-                    }
-                    catch (Exception)
+                    // If it's a device coming from the runtime, disable it.
+                    if (id != InputDevice.InvalidDeviceId)
                     {
+                        var command = DisableDeviceCommand.Create();
+                        m_Runtime.DeviceCommand(id, ref command);
                     }
+
+                    continue;
+                }
+
+                try
+                {
+                    AddDevice(m_AvailableDevices[i].description, layout, deviceId: id,
+                        deviceFlags: m_AvailableDevices[i].isNative ? InputDevice.DeviceFlags.Native : 0);
```

Comment:
Unfortunately, I think this won't work correctly. Sometime in the past, the code used to just use the layout it had before the domain reload. That was okay when we actually took all layout information across the reload.

Now, we have a setup where all layout information is considered "thrown away" during domain reloads and allowed to change completely. That, however, also means we have to do the matching process from scratch as matchers are part of what is allowed to change.

Reply:
But this new code *does* do the matching process in the same way as before, it just does it once because it calls into the new AddDevice overload, which doesn't have a TryFindMatchingControlLayout call in it :confused:

Reply:
Ah, waddayaknow, can't even read me own code. You're right.